src/robusta/core/sinks/sink_base.py

```diff
@@ -1,12 +1,29 @@
-from typing import Any
+import threading
+from abc import abstractmethod, ABC
+from collections import defaultdict
+from typing import Any, List, Dict, Tuple, DefaultDict
 
 from robusta.core.model.k8s_operation_type import K8sOperationType
 from robusta.core.reporting.base import Finding
 from robusta.core.sinks.sink_base_params import ActivityInterval, ActivityParams, SinkBaseParams
 from robusta.core.sinks.timing import TimeSlice, TimeSliceAlways
 
 
-class SinkBase:
+class SinkBase(ABC):
```

Comment:
Reviewers note: parts of the grouping functionality (mostly the setup/verification of things) are implemented in the sink base class with a view to expand the grouping mechanisms to other sinks, as it was requested in the task.